### PR TITLE
Add emphasis, strong and code support in header 

### PR DESCRIPTION
**Pull request by peng1999**
Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## Code changes

* [x] Are you doing the PR on the `next` branch?

---
This PR is a bug fix related to #569 , but the footnote problem is still unfixed due to some difficulty.

### CODE REVIEW

File: components/rendering/src/markdown.rs [hunk 1]
```diff
@@ -49,6 +49,70 @@ fn is_colocated_asset_link(link: &str) -> bool {
         && !link.starts_with("mailto:")
 }
 
+fn fix_link(link: &str, context: &RenderContext) -> Result<String> {
+    // A few situations here:
+    // - it could be a relative link (starting with `./`)
+    // - it could be a link to a co-located asset
+    // - it could be a normal link
+    // - any of those can be in a header or not: if it's in a header
```
Comment (Keats) [inline components/rendering/src/markdown.rs]: that list item is not valid anymore is it?

File: components/rendering/src/markdown.rs [hunk 2]
```diff
@@ -49,6 +49,70 @@ fn is_colocated_asset_link(link: &str) -> bool {
...
+                format!("Link {} is not valid: {}", link, res.message()).into(),
+            );
+        }
+    } else {
+        link.to_string()
+    };
+    Ok(result)
+}
+
+fn start_tag(temp_header: &mut TempHeader, tag: &Tag) -> bool {
```
Comment (Keats) [inline components/rendering/src/markdown.rs]: can we find better name than start_tag/end_tag? it's not really clear what is it doing from the signature

File: components/rendering/src/markdown.rs [hunk 3]
```diff
@@ -49,6 +49,70 @@ fn is_colocated_asset_link(link: &str) -> bool {
...
+    match tag {
+        Tag::Emphasis => temp_header.add_html("<em>"),
+        Tag::Strong => temp_header.add_html("<strong>"),
+        Tag::Code => temp_header.add_html("<code>"),
+        // Tag::Link is handled elsewhere
```
Comment (Keats) [inline components/rendering/src/markdown.rs]: possible to add where?

File: components/rendering/src/markdown.rs [hunk 4]
```diff
@@ -76,6 +140,11 @@ pub fn markdown_to_html(content: &str, context: &RenderContext) -> Result<Render
 
     {
         let parser = Parser::new_ext(content, opts).map(|event| {
+            // Trivial markup generation
```
Comment (Keats) [inline components/rendering/src/markdown.rs]: this comment doesn't really tell what the the lines below are doing

**Review by Keats — CHANGES_REQUESTED** [review]
Some nits but looks ok overall

**Comment by peng1999** [comment]
Fixed

**Comment by Keats** [comment]
Thanks!

**Pull request merged**